hydradx/tests/test_router.py

```diff
@@ -921,3 +923,47 @@ def test_buy_spot_sell_stableswap_buy_omnipool(
         raise ValueError(f"actually bought {buy_quantity} != trade size {trade_size}")
     if buy_spot != pytest.approx(buy_ex, rel=1e-08):
         raise ValueError(f"spot price {buy_spot} != execution price {buy_ex}")
+
+
+def test_calculate_buy_from_sell():
```

Comment:
This test needs to include a case where one or both assets being traded exists in at least two places in the Omnipool + stableswap ecosystem. E.g. DOT in Omnipool and DOT in a stableswap pool. Or USDT in two different stableswap pools.

Reply:
I also would want to see a test of this function in a setting where the best route for selling is not the same route as the best route for buying.

Reply:
Added those test cases.